README.md

```diff
@@ -1,2 +1,252 @@
+	console.log("NatsAsyncApiClient subscribe");
+	console.log(e);
+});
+natsClient.on(AvailableEvents.unsubscribe, (e) => {
+	console.log("NatsAsyncApiClient unsubscribe");
+	console.log(e);
+});
+natsClient.on(AvailableEvents.yield, () => {
+	console.log("NatsAsyncApiClient yield");
+});
+``` 
+
+# Template Parameters
+These are the available template parameters:
+|Parameter|Type|Description|
+|---|---|---|
+| generateTestClient | Boolean | Use this parameter to generate the [test client](#test-client). Add the following to the CLI when generating your code `--param "generateTestClient=true"`
+| promisifyReplyCallback | Boolean | Use this parameter to change from the default regular callback when using the request operation. Add the following to the CLI when generating your code `--param "promisifyReplyCallback=true"`
+
+# Test Client
+The test client is like a mirror client. It does everything the opposite, when you define a subscription operation it will generate a subscription operation in the test client. This is opposite to what the specification dictates. This client can be used to create integration tests or other useful scenarios.
+
+# Supported Content Types
+The following payload types are supported, this is limited to the underlying NATS TypeScript library:
+
+* For binary payloads use: `binary` content type
+* For json payloads use: `json` content type
+* For string payloads use: `string` content type
+
+# Client Hooks
+
+Sometimes to you want to change the data before sending or reciving it. For this purpose hooks has been added to control the flow of information outside the generated code. The hooks can be used to alter the payload before sending or after recieving any data i.e. encrypt, compress data, etc. It is possible to register as many hooks as you want however there are certain restrictions, see [hook restrictions](#Hook-restrictions).
+
+
+These are the available hooks:
+|Hookname|Callback type|Description|
+|---|---|---|
+| BeforeSendingData | (messageToSend: any) => any | Called before sending any data. 
+| receivedData | (receivedData: any) => any | Hook is called after data is received before the client tries to peace the message back together.
+
+## Node.js example
+This example uses the msgpack library to encode and decode the message before sending and receiving data.
+```js
+const {AvailableEvents, NatsAsyncApiClient, Hooks} = require("nats-client");
+const natsClient = new NatsAsyncApiClient();
+const msgpack = require("msgpack-lite");
+function encode(msg){
+	console.log("encode");
+	console.log(msg);
+	// encode from JS Object to MessagePack (Buffer)
+	return msgpack.encode(msg);
+}
+function decode(msg){
+	// decode from MessagePack (Buffer) to JS Object
+	console.log("Decoding");
+	console.log(msg);
+	return msgpack.decode(msg.data); // => {"foo": "bar"}
+}
+Hooks.getInstance().registerBeforeSendingData(encode);
+Hooks.getInstance().registerreceivedData(decode);
+``` 
+## Hook restrictions
```

Comment:
It took me a while to realize you're not talking about Generator hooks here 😄

Reply:
Yea no, that is the generated client's hooks 😄